visualizer/tooltip.js

```diff
@@ -87,7 +87,13 @@ class Tooltip extends HtmlContent {
     this.d3TooltipInner.classed('top bottom', false)
     this.d3TooltipInner.classed(verticalAlign, true)
 
-    let innerRect = Object.assign({ }, targetRect || d3TargetElement.node().getBoundingClientRect())
+    const obj = targetRect || d3TargetElement.node().getBoundingClientRect()
+    let innerRect = {
+      x: obj.x,
+      y: obj.y,
+      width: obj.width,
+      height: obj.height
+    }
```

Comment:
Both `obj` and `innerRect` don't seem to do anything except hold those four properties for the scope of this function, so we could make it cleaner and clearer with:

```
   let {
      x,
      y,
      width,
      height
    } = targetRect || d3TargetElement.node().getBoundingClientRect()
```

...then replace `innerRect.x` with `x`, `innerRect.width` with `width` etc.

Reply:
Hmm.. I didn't mind scoping those variables but I guess it's just a personal habit.